src/app/onion/core/learning-object.service.ts

```diff
@@ -136,6 +126,28 @@ export class LearningObjectService {
       )
       .toPromise();
   }
+
+  /**
+   * Unpublish a learning object
+   * @param {learningObject} learningObject the learning object to be unpublished
+   */
+  unpublish(learningObject: LearningObject) {
+    const route = USER_ROUTES.UNPUBLISH_LEARNING_OBJECT(
+      this.auth.user.username,
+      learningObject.name
+    );
+
+    return this.http
+      .patch(
+        route,
+        {
+          id: learningObject.id,
+          published: !learningObject.published
```

Comment:
If this method is to only unpublish a learning object, should we use `published: false` in the patch request instead of `published: !learningObject.published`?